### PR TITLE
CORTX-30932: Adding view class and routes to REST API framework

### PR DESCRIPTION
**Pull request by gargpalak048**
# Problem Statement
- Adding view class and routes to REST API framework

# Design
-  Added a View class and routes to REST API Framework
- Added resource Rest API for HA Framework

# Coding
-  [x] Coding conventions are followed and code is consistent

# Testing 
- [x] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM
https://jts.seagate.com/secure/attachment/515451/test_results_30932.txt
# Review Checklist 
- [x] PR is self reviewed
- [x] JIRA number/GitHub Issue added to PR
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained
- [ ] Is there a change in filename/package/module or signature? [Y/N]: 
- [ ] If yes for above point, is a notification sent to all other cortx components? [Y/N]
- [ ] Side effects on other features (deployment/upgrade)? [Y/N]
- [ ] Dependencies on other component(s)? [Y/N]
-     If yes for above point, post link to the corresponding PR.

# Review Checklist 
- [ ] Is perfline test run and the report with and without the changes updated in the PR? [Y/N]: 

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: ha/k8s_setup/ha_setup.py [hunk 1]
```diff
@@ -220,6 +242,7 @@ def process(self):
             health_comm_msg_type = FAULT_TOLERANCE_KEYS.MONITOR_HA_MESSAGE_TYPE.value
 
             conf_file_dict = {'LOG' : {'path' : ha_log_path, 'level' : const.HA_LOG_LEVEL},
+                         'service_config' : {'endpoint' : ha_endpoint},
```
Comment (gargpalak048) [inline ha/k8s_setup/ha_setup.py:245]: same comment as above:
https://github.com/Seagate/cortx-ha/pull/701/files#r858717510

**Review by gauri-bhosale — COMMENTED** [review]
please resolve codecy issue

File: ha/fault_tolerance/resources.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+
+# CORTX Python common library.
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
```
Comment (mariyappanp) [inline ha/fault_tolerance/resources.py:4]: Please change it to 2022

File: ha/fault_tolerance/rest_api.py [hunk 3]
```diff
@@ -0,0 +1,236 @@
+import signal
```
Comment (mariyappanp) [inline ha/fault_tolerance/rest_api.py:1]: Copyright is missing

File: ha/fault_tolerance/routes.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
```
Comment (mariyappanp) [inline ha/fault_tolerance/routes.py:3]: Year correction is required.

File: ha/fault_tolerance/view.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+
+# CORTX Python common library.
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
```
Comment (mariyappanp) [inline ha/fault_tolerance/view.py:4]: Please change the year

**Comment by shailesh-vaidya** [comment]
Closing as an obsolete

**Pull request closed without merging**